File: DotNet/SETPractice/DesignPattern/BuilderPattern/DieselCar.cs
namespace DesignPattern.BuilderPattern
{
    public class DieselCar
    {
        public ICarBuilder CarBuilder { get; set; }

        public DieselCar(ICarBuilder carBuilder)
        {
            CarBuilder = carBuilder;
        }

        public Car AssembleCar()
        {
            CarBuilder.SetBody();
            CarBuilder.SetSeats(7);
            CarBuilder.SetHeadLights(4);
            CarBuilder.SetDieselEngine();
            return CarBuilder.GetCar();
        }
    }
}

File: Java/SETPractice/src/main/java/DesignPattern/BuilderPattern/DieselCar.java
package DesignPattern.BuilderPattern;

public class DieselCar {
    public ICarBuilder carBuilder;

    public DieselCar(ICarBuilder carBuilder) {
        this.carBuilder = carBuilder;
    }

    public Car AssembleCar() {
        carBuilder.SetBody();
        carBuilder.SetDieselEngine();
        carBuilder.SetHeadLights(4);
        carBuilder.SetSeats(7);
        return carBuilder.GetCar();
    }
}
